src/Symfony/Component/HttpFoundation/HeaderBag.php

```diff
@@ -103,28 +111,21 @@ public function add(array $headers)
      *
      * @param string      $key     The header name
      * @param string|null $default The default value
-     * @param bool        $first   Whether to return the first value or all header values
      *
-     * @return string|string[]|null The first header value or default value if $first is true, an array of values otherwise
+     * @return string|null The first header value or default value if $first is true, an array of values otherwise
      */
-    public function get($key, $default = null, $first = true)
+    public function get($key, $default = null)
     {
-        $key = str_replace('_', '-', strtolower($key));
-        $headers = $this->all();
+        $headers = $this->all((string) $key);
+        if (2 < \func_num_args()) {
+            @trigger_error(sprintf('Passing $first as a third argument is deprecated since Symfony 4.4. %s will return the first header value, use %s::all($key) instead to get an array of values.', __METHOD__, __CLASS__), E_USER_DEPRECATED);
```

Comment:
This sentence reads strangely: "first" being the *third* argument doesn't play well. What about:
`Passing a third argument to "%s()" is deprecated since Symfony 4.4, use method "all()" instead.`